ExtremeValueProject/TypeOfCDF.lean

```diff
@@ -183,7 +184,117 @@ lemma not_tendsto_cdf_of_expanding_of_tendsto_not_isDegenerate
     (hG : ¬ G.IsDegenerate) {A : ℕ → AffineIncrEquiv}
     (a_lim : Tendsto (fun n ↦ (A n).val.toAffineMap.coefs_of_field.1) atTop atTop) :
     ¬ ∀ x, ContinuousAt G' x → Tendsto (fun n ↦ ((A n) • (F n)) x) atTop (𝓝 (G' x)) := by
-  sorry -- **Issue #40**
+  intro nottrue
+  have ⟨x1,x2,x1_lt_x2,Gx1_pos,Gx2_bound,x1_cont,x2_cont⟩:=
+    CumulativeDistributionFunction.exists₂_continuousAt_of_not_isDegenerate _ hG
+  have right_tendsto {z : ℝ} (z_spec_cont : ContinuousAt G' z) {s : ℕ → ℕ}
+      (s_atTop : Tendsto s atTop atTop) :
+      Tendsto (fun k ↦ (A (s k) • F (s k)) z) atTop (𝓝 (G' z)) := by
+    change Tendsto ((fun n ↦ (A n • F n) z) ∘ s) atTop (𝓝 (G' z))
+    have z_converge := nottrue z z_spec_cont
+    unfold Tendsto at z_converge ⊢
+    refine le_trans ?_ z_converge
+    exact fun ⦃u⦄ ↦ (s_atTop ·)
+  have left_tendsto {x1 : ℝ} (x1_cont : ContinuousAt G x1) {s : ℕ → ℕ}
+      (s_atTop : Tendsto s atTop atTop) :
+      Tendsto (fun k ↦ F (s k) x1) atTop (𝓝 (G x1)) := by
+    unfold Tendsto
+    have x1_tendsto:= F_lim _ x1_cont
+    refine le_trans ?_ x1_tendsto
+    rw [(by rfl : (fun n ↦ F (s n) x1) = (fun n ↦ F n x1) ∘ s)]
+    exact fun ⦃u⦄ ↦ (s_atTop ·)
+  -- the names `lt` and `min` have lost their meaning.
+  have not_bounded_after' {B : ℕ → ℝ} {lt : ℝ → ℝ → Prop} {min : ℝ → ℝ → ℝ}
+      (not_bounded : ∀z, ∃ x, lt (B x) z )
+      (min_spec : ∀ ⦃a b c⦄, lt a (min b c) → a ≠ b ∧ lt a c)
+      (z) (t) : ∃ x ≥ t, lt (B x) z := by
+    induction t generalizing z with
+    | zero =>
+      simp only [ge_iff_le, zero_le, true_and]
+      exact not_bounded z
+    | succ t prev =>
+      -- `prev (min (B t) z)` ensures that `y ≠ t`, using `B y < B t`
+      have ⟨y, y_gt_t, y_spec⟩ := prev (min (B t) z)
+      apply min_spec at y_spec
+      have yyt : t ≠ y := by
+        intro con
+        exact (con ▸  y_spec.left) rfl
+      refine ⟨y, Nat.lt_iff_add_one_le.mp (Nat.lt_of_le_of_ne y_gt_t yyt), y_spec.right⟩
+  have ⟨below,claim_below⟩ : ∃ below, ∀ n, A n x1 > below := by
+    by_contra not_bounded
+    simp only [gt_iff_lt, not_exists, not_forall, not_lt] at not_bounded
+    have not_bounded_after := not_bounded_after' not_bounded
+      (lt := (· ≤ ·)) (min := (fun a b ↦ min (a - 1) b))
+      (by intro a b c abc ; constructor <;> linarith [le_inf_iff.mp abc] )
+    have ⟨z,z_spec_cont,z_spec_lt⟩ : ∃z, ContinuousAt G' z ∧ G' z < G x1 := by
+      have ⟨z,_,_,z_lt,_,z_cont,_⟩:= G'.forall_pos_exists_lt_gt_continuousAt Gx1_pos
+      use z
```

Comment:
You can `obtain` directly. 

```suggestion
    obtain ⟨z, _, _, z_spec_lt, _, z_spec_cont, _⟩ := G'.forall_pos_exists_lt_gt_continuousAt Gx1_pos
```

Also `obtain` is considered more appropriate for "data" (terms of type other than a `Prop`). The `have` tactic is meant to be used for `Prop`s. (This small style comment applies to many `have`s in  your code.)